src/markdown/transformer.ts

```diff
@@ -87,9 +88,19 @@ export class Transformer {
   }
 
   calculateDeckName(generatedName: string | null = null): string {
-    return this.useDefault
-      ? this.defaultDeck
-      : generatedName || this.defaultDeck;
+    if (this.strategy === DeckNameStrategy.useDefault) {
+      return this.defaultDeck;
+    } else if (this.strategy === DeckNameStrategy.ParseTitle) {
+      return generatedName || this.defaultDeck;
+    } else {
+      const rootPath = workspace.workspaceFolders?.[0].uri.path;
+      const filePath = window.activeTextEditor?.document.uri.path;
```

Comment:
Im ok with this but something which may be more robust..

You can get the workspace folder above by passing in the uri from here.
https://code.visualstudio.com/api/references/vscode-api#workspace -- getWorkspaceFolder instead of indexing the [0] one

So passing the filePath url should always give back the correct workspace.